tests/storage/cbt/pull_collect_runner.py

```
@@ -0,0 +1,190 @@
+    scan_offset = 0
+    disk_size_bytes = int(params["disk_size_bytes"])
+    while scan_offset < disk_size_bytes:
+        scan_length = min(PULL_MAP_SCAN_LIMIT_BYTES, disk_size_bytes - scan_offset)
+        map_url = (
+            f"{params['map_endpoint']}?x-kubevirt-export-token={params['export_token']}"
+            f"&offset={scan_offset}&length={scan_length}"
+        )
+        map_payload = json.loads(_curl_pull_endpoint(endpoint_url=map_url))
+        for raw_extent in map_payload["extents"]:
+            if str(raw_extent["description"]).lower() not in PULL_MAP_HOLE_DESCRIPTIONS:
+                data_extents.append((int(raw_extent["offset"]), int(raw_extent["length"])))
```

Ruff (0.15.21)

[warning] Use `list.extend` to create a transformed list

(PERF401)